genomelake/extractors.py

```diff
@@ -121,18 +122,18 @@ def _get_output_shape(num_intervals, width):
         return (num_intervals, width)
 
     @staticmethod
-    def _bigwig_extractor(datafile, intervals, out=None, **kwargs):
+    def _bigwig_extractor(bw, intervals, out=None, **kwargs):
         nan_as_zero = kwargs.get('nan_as_zero', True)
         if out is None:
             width = intervals[0].stop - intervals[0].start
             out = np.zeros((len(intervals), width), dtype=np.float32)
-
-        bw = pyBigWig.open(datafile)
+            
         for index, interval in enumerate(intervals):
             out[index] = bw.values(
                 interval.chrom, interval.start, interval.stop)
             if nan_as_zero:
                 nan_to_zero(out[index])
-        bw.close()
-
         return out
+    
+    def close(self):
```

Comment:
I don't think it will matter for typical usage, but you might consider adding a `__del__` function that just calls this within a "try/except pass" block (https://docs.python.org/3/reference/datamodel.html#object.__del__)

Reply:
I'm thinking of scenarios where someone is running genomelake in a training loop, and keeps instantiating extractors, e.g. for each epoch of training a model, without explicitly calling close(). Adding the `__del__` would stop us from filling up the file descriptor table.

Reply:
will add that